gammapy/maps/base.py

```diff
@@ -101,6 +102,57 @@ def create(cls, **kwargs):
         else:
             raise ValueError('Unrecognized map type: {}'.format(map_type))
 
+    @classmethod
+    def read(cls, filename, **kwargs):
+        """Read a map from a FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the FITS file.
+        hdu : str
+            Name or index of the HDU with the map data.
+        hdu_bands : str
+            Name or index of the HDU with the BANDS table.  If not
+            defined this will be inferred from the FITS header of the
+            map HDU.
+
+        Returns
+        -------
+        out_map : `~MapBase`
+            Map object
+
+        """
+        with fits.open(filename) as hdulist:
+            out_map = cls.from_hdulist(hdulist, **kwargs)
+        return out_map
+
+    def write(self, filename, **kwargs):
+        """Write to a FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            Output file name.
+        extname : str
+            Set the name of the image extension.  By default this will
+            be set to SKYMAP (for BINTABLE HDU) or PRIMARY (for IMAGE
+            HDU).
+        extname_bands : str
+            Set the name of the bands table extension.  By default this will
+            be set to BANDS.
+        hpxconv : str
+            Format convention for HEALPix maps.  This option can be used to
+            write files that are compliant with non-standard HEALPix
+            conventions.
+        sparse : bool
+            Sparsify the map by dropping pixels with zero amplitude.
+
+        """
+        hdulist = self.to_hdulist(**kwargs)
+        overwrite = kwargs.get('overwrite', True)
+        hdulist.writeto(filename, overwrite=overwrite)
```

Comment:
There's a test fail on Windows:
```
 WindowsError: [Error 32] The process cannot access the file because it is being used by another process
```
https://ci.appveyor.com/project/cdeil/gammapy/build/1.0.2536/job/ej6xbtb9bpu7j7g1#L1570

Looking at http://docs.astropy.org/en/stable/_modules/astropy/io/fits/hdu/hdulist.html#HDUList.writeto I see that "close" is already called.
So to me, your I/O calls look right and I had hoped that this way every `read` and `write` call would never leave files open. But apparently that's not the case on Windows? (Or there's an issue where different pytest tests are running on the same test files in parallel. I don't think that's the case, but I'm not sure).

I'm not really sure what's going on / how to resolve this. Maybe mark the test as xfail on Windows so that this can go in, and then investigate / fix the Windows issue later, possibly asking Windows / Astropy people for help.